### ISSUE TITLE
ASW_cellType and iASW use predicted cluster labels rather than the celltype column

### ISSUE THREAD

**Issue by dnncha** [PYangLab/scMultiBench#4]
The embedding evaluator passes `label_key="cluster"` to both silhouette metrics, although it has already loaded the biological labels into `obs["celltype"]`. The ordinary silhouette result is then printed as `ASW_cellType`.

At commit `0c68f87d554a52e363dbfda2a9d1fa9538eeae9f`, [evaluation_pipelines/scib_metrics/scib_metrics.py](https://github.com/PYangLab/scMultiBench/blob/0c68f87d554a52e363dbfda2a9d1fa9538eeae9f/evaluation_pipelines/scib_metrics/scib_metrics.py#L86-L100) contains:

```python
asw = scib.me.silhouette(adata_integrated, label_key="cluster", embed="X_emb")
print("ASW_cellType:", asw)
iasw = scib.me.isolated_labels_asw(adata_integrated, batch_key="batch", label_key="cluster", embed="X_emb", iso_threshold=num)
```

In the same file, ARI, NMI and cLISI use `celltype` as the biological reference. `cluster` is read from the supplied `cluster_leiden` dataset and is not an alias for `celltype`.

The [paper's Evaluation metrics section](https://www.nature.com/articles/s41592-025-02856-3) describes ASW by cell type. The current call instead measures the compactness/separation of the inferred clusters. That can be a valid internal clustering criterion, but it does not measure separation of the supplied cell types. Is this intentional, or should these two calls use `label_key="celltype"`?

## Executed numerical example

I constructed two embeddings of the same 120 observations and the same two balanced cell types. Both are clustered with actual scikit-learn KMeans, with two clusters and fixed initialization settings.

- Biological embedding: the two groups correspond to the cell types.
- Technical embedding: the two compact groups each contain equal numbers of both cell types. The grouping is unrelated to the biological labels.

| Candidate | Released ASW call (`cluster`) | Cell-type ASW (`celltype`) | ARI: inferred clusters vs cell types |
|---|---:|---:|---:|
| Biological separation | 0.9192562199 | 0.9192562199 | 1.0000000000 |
| Technical separation, mixed cell types | **0.9963883078** | **0.4918026995** | -0.0084745763 |

The current call favours the technical embedding. Changing only the label source reverses that metric's preference. In this constructed example, the technical groups do not separate the biological labels at all.

These are **synthetic metric comparisons**, not the paper's embedding outputs, published method ranks, or a corrected aggregate leaderboard. The construction deliberately contrasts biological and non-biological separation; it does not estimate the frequency or effect size in the original study.

## What was executed

I recovered the complete 5,723-byte benchmark script and verified its Git blob SHA as `b8549d293d2a6eaef9af91bd7842dc68e15e9fcf`. The test extracts and executes its original `asw = ...` assignment using Python's AST. The corrected execution changes only the `label_key` argument.

The numerical silhouette body is taken from [scib v1.1.5](https://github.com/theislab/scib/blob/v1.1.5/scib/metrics/silhouette.py) and calls real scikit-learn. A SimpleNamespace containing real pandas/NumPy objects supplies `.obs` and `.obsm`; distances, clustering and scores are not mocked. I have not run the full scMultiBench CLI, native AnnData/scanpy integration or the paper's original environments. The scib reference version is an explicit audit dependency, not a claim about the version used in the publication.

Across all 30 tested seeds, the two metric preferences reverse. The suite also passed 120 independent brute-force silhouette comparisons, 120 row-permutation comparisons, 120 label-renaming comparisons, and 120 common rotation/translation/scaling comparisons. All tested seeds are retained in the results, not only seed zero.

Environment: Python 3.13.5, NumPy 2.3.5, SciPy 1.17.0, pandas 2.2.3. The standalone example below uses the same fixture and the same normalized silhouette calculation; scikit-learn's exact installed version is recorded in the downloadable execution record prepared with the report.

```python
import numpy as np
from sklearn.cluster import KMeans
from sklearn.metrics import adjusted_rand_score, silhouette_score
from threadpoolctl import threadpool_limits

n = 120
truth = np.repeat([0, 1], n // 2)
technical_partition = np.tile([0, 1], n // 2)
rng = np.random.default_rng(0)
biological = rng.normal(0, .35, (n, 2)) + np.column_stack(
    (truth * 4. - 2., np.zeros(n)))
technical = rng.normal(0, .08, (n, 2)) + np.column_stack(
    (technical_partition * 20. - 10., np.zeros(n)))

values = []
with threadpool_limits(limits=1):
    for name, x in [('biological', biological), ('technical', technical)]:
        clusters = KMeans(n_clusters=2, random_state=0, n_init=10).fit_predict(x)
        released = (silhouette_score(x, clusters) + 1) / 2
        celltype = (silhouette_score(x, truth) + 1) / 2
        ari = adjusted_rand_score(truth, clusters)
        print(name, released, celltype, ari)
        values.append((released, celltype))
assert values[1][0] > values[0][0]
assert values[1][1] < values[0][1]
```

For comparison, a cell-type silhouette score should not depend on which predicted clustering is supplied when the embedding and biological labels are unchanged. The current call does, and changing the biological labels alone leaves its result unchanged.

## Proposed change

If cell-type separation is the intended quantity, change the two `label_key="cluster"` arguments above to `label_key="celltype"`. That is a two-line patch; it leaves clustering, ARI/NMI, specificity and F1 unchanged. It is not a complete correction of every metric in this evaluator.

There is a related isolated-label policy question: `read_batch` numbers batches from 1, but `num = np.max(batch)+1` is passed as `iso_threshold`. For B batches this gives B+1. scib defines this threshold as the maximum number of batches containing a label, so every observed label qualifies. In a separate three-batch fixture, a label confined to one batch and two labels present in all three are all selected at threshold 4; scib's default selects only the one-batch label. I executed the reference `get_isolated_labels` numerical body to check this. The graph evaluator passes the same threshold to isolated-label F1. Could you clarify the intended isolation rule? I have not changed that policy in the proposed two-line patch.

To quantify publication effects, the required inputs are the original embeddings, biological labels, inferred cluster labels, and intended isolation policy. The current result establishes the label-source mismatch and a possible ASW metric preference reversal, not a changed original-paper recommendation.

**Comment by dnncha** [comment]
The report, two-line patch and compact runnable example are now public here:
https://github.com/dnncha/Dnncha/tree/main/research/scmultibench-silhouette

The public runner reproduced the values above for all 30 designed seeds and passed 120 independent direct-formula comparisons:
https://github.com/dnncha/Dnncha/actions/runs/34127150677

That runner executes the compact sklearn reproduction, not the full scMultiBench CLI. The report distinguishes it from the local AST-extracted upstream-call test and records the remaining native-pipeline and original-data gaps. These values are normalized silhouettes, not classification accuracies.

**Comment by dnncha** [comment]
I have now done two follow-ups on the complete released embedding demo used in the evaluator's example command.

**1. Native scib confirmation.** The earlier synthetic label-source result reproduces with actual scib 1.1.5 and AnnData 0.10.9 after verifying the complete scMultiBench evaluator and complete scib `silhouette.py` / `isolated_labels.py` blobs. The public run is https://github.com/dnncha/Dnncha/actions/runs/34139782478.

**2. Released demo reanalysis.** I downloaded and Git-blob-verified the three released label CSVs, `embedding.h5`, and `sinfonia_clustering.h5` at commit `0c68f87d...`. The data contain 28,574 observations, 15 embedding dimensions, 3 batches, 28 biological cell types and 28 inferred clusters.

On the same released embedding:

- released ordinary call (`label_key="cluster"`): **0.614295482635498**
- ordinary ASW using biological cell types: **0.5696005523204803**

So the label-source mismatch changes the measured ASW by about 0.044695 on this released example, not only on the constructed fixture.

For iASW, the released call also has the independent threshold issue now tracked in #6. Decomposing both choices on the same released data gives:

- released cluster labels + threshold 4: **0.6303640834438349**
- cell-type labels + threshold 4: **0.5768824066368065**
- cell-type labels + scib's default isolated-label policy: **0.86563840508461**

The last value is driven by the single cell type present in the minimum number of batches (`Erythroid `, one batch). I am not calling that a corrected paper value until the intended isolation definition and publication environment are confirmed.

Report, source-verified scripts and durable summary:
https://github.com/dnncha/Dnncha/tree/2c068640f9b32386b9573790765e56c837d8a537/research/scmultibench-scib-native

The CI artifact includes per-observation silhouette samples and both full result records. Its downloaded SHA-256 is `4d038bee0a0795596c2f3d6af61de588944dcdc2db137035fdfcb33546d7035c`.

These repository files are released demo inputs; I have not established that they are a named original-publication method output or reconstructed the overall leaderboard.
